app/components/on-this-page.tsx

```diff
@@ -17,7 +35,12 @@ export function LargeOnThisPage({ doc }: { doc: Doc }) {
               dangerouslySetInnerHTML={{
                 __html: heading.html || "",
               }}
-              className="block py-1 text-sm text-gray-400 hover:text-gray-900 active:text-red-brand dark:text-gray-400 dark:hover:text-gray-50 dark:active:text-red-brand"
+              className={
+                (activeHeading == heading.slug
+                  ? "text-gray-900 dark:text-gray-50"
+                  : "") +
+                " block py-1 text-sm text-gray-400 hover:text-gray-900 active:text-red-brand dark:text-gray-400 dark:hover:text-gray-50  dark:active:text-red-brand"
+              }
```

Comment:
Small nit, but add this import

```ts
import classNames from "classnames";
```

and we can update to the following

```suggestion
              className={classNames(
                activeHeading == heading.slug &&
                  "text-gray-900 dark:text-gray-50",
                "block py-1 text-sm text-gray-400 hover:text-gray-900 active:text-red-brand dark:text-gray-400 dark:hover:text-gray-50  dark:active:text-red-brand"
              )}
```

Reply:
Done